istioctl/cmd/describe.go

```diff
@@ -1186,3 +1197,94 @@ func containerReady(pod *v1.Pod, containerName string) (bool, error) {
+// When the client doesn't have access to root namespace, it will only show workload namespace Peerauthentications.
+func describePeerAuthentication(writer io.Writer, configClient istioclient.Interface, istioNamespace, workloadNamespace string, podsLabels k8s_labels.Set) error { // nolint: lll
+	workloadPAList, err := configClient.SecurityV1beta1().PeerAuthentications(workloadNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch workload namespace PeerAuthentication: %v", err)
+	}
+
+	hasRootNamespaceAccess := true
+	rootPAList, err := configClient.SecurityV1beta1().PeerAuthentications(istioNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		hasRootNamespaceAccess = false
+		fmt.Fprintf(writer, "Cannot analyze active mTLS mode since failed to fetch root namespace PeerAuthentication: %v:\n", err)
+	}
+
+	allPAs := append(rootPAList.Items, workloadPAList.Items...)
+	if len(allPAs) == 0 {
+		return nil
+	}
+
+	var cfgs []*config.Config
+	for _, pa := range allPAs {
+		pa := pa
+		cfg := crdclient.TranslateObject(&pa, config.GroupVersionKind(pa.GroupVersionKind()), "")
+		cfgs = append(cfgs, &cfg)
+	}
+
+	matchedPA := findMatchedConfigs(podsLabels, cfgs)
+	printConfigs(writer, matchedPA)
+
+	if hasRootNamespaceAccess {
+		activePA := authnv1beta1.ComposePeerAuthentication(istioNamespace, matchedPA)
+		printPeerAuthentication(writer, activePA)
+	}
+
+	return nil
+}
+
+// Workloader is used for matching all configs
+type Workloader interface {
+	GetSelector() *typev1beta1.WorkloadSelector
+}
+
+// findMatchedConfigs should filter out unrelated configs that are not matched given podsLabels.
+// Configs passed into this method should only contains workload's namespaces configs
+// and rootNamespaces configs, caller should be responsible for controlling configs passed
+// in.
+func findMatchedConfigs(podsLabels k8s_labels.Set, configs []*config.Config) []*config.Config {
+	var cfgs []*config.Config
+
+	for _, cfg := range configs {
+		cfg := cfg
+		spec := cfg.Spec.(Workloader)
+		if spec.GetSelector() == nil || len(spec.GetSelector().MatchLabels) == 0 {
+			cfgs = append(cfgs, cfg)
+		} else {
+			selector := spec.GetSelector().GetMatchLabels()
+			slr := k8s_labels.SelectorFromSet(selector)
+			if slr.Matches(podsLabels) {
+				cfgs = append(cfgs, cfg)
+			}
+		}
+	}
+
+	return cfgs
+}
+
+func printConfigs(writer io.Writer, configs []*config.Config) {
+	if len(configs) == 0 {
+		return
+	}
+	fmt.Fprintf(writer, "Applied %s:\n", configs[0].Meta.GroupVersionKind.Kind)
+	var cfgNames string
+	for _, cfg := range configs {
+		cfgNames += cfg.Meta.Name + "." + cfg.Meta.Namespace + ", "
+	}
+	fmt.Fprintf(writer, "   %s\n", cfgNames)
+}
+
+func printPeerAuthentication(writer io.Writer, pa *v1beta1.PeerAuthentication) {
+	fmt.Fprintf(writer, "Active PeerAuthentication:\n")
+	fmt.Fprintf(writer, "   Workload mTLS: %s\n", pa.Mtls.Mode.String())
```

Comment:
It's worth to also print the namespace/name of the active policy.

Reply:
The composed PA might not be from only one PA, to print all the active PA would involve bigger change, probably not reuse ComposePeerAuthentication(). And that would have potential sync problem with istiod's logic. 
Thats why end up printing out all applied PAs and show the active mode. The method name might be misleading though, will change that.

Reply:
yeah, that makes sense, we can add this later.